librosa/util/utils.py

```diff
@@ -627,8 +632,19 @@ def normalize(S, norm=np.inf, axis=0):
     else:
         raise ParameterError('Unsupported norm: {}'.format(repr(norm)))
 
+    # get the vectors-ids smaller than the threshold
+    small_values_idx = length < threshold
+
     # Avoid div-by-zero
-    length[length < SMALL_FLOAT] = 1.0
+    length[small_values_idx] = 1.0
+
+    if axis == 1:
```

Comment:
My usual trick for this goes like the following:

```python
idx = [slice(None)] * ndim
idx[axis] = target_axis_indexing_slice
f(x[idx])
```

You can see this in action [here](https://github.com/bmcfee/librosa/blob/master/librosa/util/utils.py#L343-L351)